src/cplscheme/config/CouplingSchemeConfiguration.cpp

```diff
@@ -916,6 +894,20 @@ CouplingSchemeConfiguration::getTimesteppingMethod(
   }
 }
 
+void CouplingSchemeConfiguration::checkIterationLimits() const
+{
+  if (_config.convergenceMeasureDefinitions.empty()) {
+    PRECICE_CHECK(_config.maxIterations != -1,
+                  "No convergence measures without a maximum iteration limit leads to endless iterations. "
+                  "Please define a convergence measure or set a maximum iteration limit using <max-iterations value=\"...\" />.");
```

Comment:
```suggestion
                  "Not defining convergence measures without providing a maximum iteration limit is forbidden."
                  "Please define a convergence measure or set a maximum iteration limit using <max-iterations value=\"...\" />.");
```
I think giving the reason here is a bit an overkill.